src/com/musala/xml/parser/Main.java

```diff
@@ -0,0 +1,181 @@
+        File configFile = new File(classload.getResource(pathname3).getFile());
+        JAXBContext jaxb = JAXBContext.newInstance(School.class);
+        Unmarshaller jaxbUnmarsheller = jaxb.createUnmarshaller();
+        School mySchool = (School) jaxbUnmarsheller.unmarshal(configFile);
+        System.out.println(mySchool.getStudents().getStudent().get(0).getAge());
+        System.out.println(mySchool.getStudents().getStudent().get(1).getAge());
+        System.out.println(mySchool.getSchoolClasses().get(0).getTeacherReference());
+        System.out.println(mySchool.getTeachers().getTeacher().get(0).getAge());
+        Print(mySchool);
+    }
+
+private static void Print(School myschool){
+    int numberOfClasses=myschool.getSchoolClasses().size();
+   System.out.printf("The school name is: %s!%n There %s %d school classes.%n",myschool.getName(),numberOfClasses>1?"are":"is",numberOfClasses);
+
+for (int i = 0; i < numberOfClasses; i++) {
+    String classNum=classTransferator(i+1);
+    System.out.printf("The %s class have %d students: %n",classNum,myschool.getSchoolClasses().get(i).getStudentsReference().size());
+   for (int j = 0; j <myschool.getSchoolClasses().get(i).getStudentsReference().size(); j++) {
+       String currentReference=myschool.getSchoolClasses().get(i).getStudentsReference().get(j);
+    Student currentStudent= (Student)searchTheStudentByReference(currentReference, myschool,"Student");
+   System.out.printf("%s is %d years old with fac. number %s.%n",currentStudent.getName(),currentStudent.getAge(),currentStudent.getFacultyNumber());
+   }
+   
+   String currentReference=myschool.getSchoolClasses().get(i).getTeacherReference();
+   Teacher currentTeacher= (Teacher)searchTheStudentByReference(currentReference, myschool,"Teacher");
+   System.out.printf("The teacher of the class is %s- %d years old",currentTeacher.getName(),currentTeacher.getAge());
+}
+   
+}            
+ 
+
+
+    private static Object searchTheStudentByReference(String studentReference, School myschool, String typeOfPerson) {
+        switch (typeOfPerson) {
+            case "Student":
+                for (int i = 0; i < myschool.getStudents().getStudent().size(); i++) {
+                    if (studentReference.equals(myschool.getStudents().getStudent().get(i).getReferenceNumber())) {
+                        return myschool.getStudents().getStudent().get(i);
+                    }
+                }
+                return null;
+
+            case "Teacher": {
+                for (int i = 0; i < myschool.getTeachers().getTeacher().size(); i++) {
+                    if (studentReference.equals(myschool.getTeachers().getTeacher().get(i).getReferenceNumber())) {
+                        return myschool.getTeachers().getTeacher().get(i);
+                    }
+                }
+                return null;
+            }
+        }
+
+        return null;
+    }
+
+    private static String classTransferator(int numberOfClass) {
```

Comment:
"classTransferator" is suitable name for a class, not method. A method's name should reflect the action it performs. Something like transformClassNumberToString(int numberOfClass) will be perfect.